pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -150,12 +162,58 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		reasonsForWorkerPoolKubernetesUpdate[w.Name] = reasonForWorkerPoolKubernetesUpdate
 	}
 
-	maintainOperation(shoot)
+	for workerPool, reason := range reasonsForWorkerPoolKubernetesUpdate {
+		if reason != "" {
+			operation := fmt.Sprintf("For worker pool %s: %s", workerPool, reason)
+			operations = append(operations, operation)
+		}
+	}
+
+	operation := maintainOperation(maintainedShoot)
+	if operation != "" {
+		operations = append(operations, "Other task started during maintenance time - "+operation)
+	}
+
+	if operations != nil {
+		patch := client.MergeFrom(shoot.DeepCopy())
+		shoot.Status.LastMaintenance = &gardencorev1beta1.LastMaintenance{
+			Description:   strings.Join(operations, ", "),
+			TriggeredTime: metav1.Time{Time: r.Clock.Now()},
+		}
+
+		// First dry run the update call to check if it can be executed successfully.
+		// If not shoot maintenance is marked as failed and is retried only in
+		// next maintenance window.
+		if err := r.Client.Update(ctx, maintainedShoot, &client.UpdateOptions{
+			DryRun: []string{metav1.DryRunAll},
+		}); err != nil {
+			if hasMaintainNowAnnotation(shoot) {
+				delete(shoot.Annotations, v1beta1constants.GardenerOperation)
+			}
```

Comment:
Why do we do this when we don't update the shoot but return in L200? This would not be reflected in the system, would it?

Reply:
> This would not be reflected in the system, would it?

Yes, patch call on status updates the annotation.

>Why do we do this when we don't update the shoot but return in L200? 

Removing the annotation is required. For ex- if the maintenance fails and later the reason for failure is fixed and if maintenance is triggered again with annotation it will fail with the reason that annotation is already present.